### PR TITLE
Update ihp-sg13g2 parasitics rules

### PR DESCRIPTION
**Pull request by Syndace**
I'm pretty sure the rcx rules for ihp-sg13g2 currently included in this repo are just placeholders rather than actually extracted ones. I grew suspicious because hundreds of hold-time violations were reported for our design after parasitics extraction, even though the design was violation-free just before detailed routing.

The rules I am proposing in this pr are ones I've extracted a year ago when we taped out at IHP with OpenLane, using a proprietary PDK for the sg13g2 process node. Even though the rules were extracted for a different PDK, the process is the same, thus parasitics information should be portable.

@KrzysztofHerman I'd like your opinion on this one.

Off-topic: we taped out on Monday! In a few months we'll see whether ORFS & IHP's open PDK already work together to produce working chips :)

### CODE REVIEW

**Comment by maliberty** [comment]
Congratulations on the tapeout!

**Comment by tspyrou** [comment]
It would be nice if you would let us know what kind of device you taped out. We could work on a blog entry too if you are open to that.

**Comment by mi-and-core** [comment]
I'm not involved, but it seems to be part of the "VE-HEP" project ("Hochschule RheinMain", Germany)

https://hep-alliance.org/Project/

https://riscv.org/news/2023/10/research-consortium-sets-standards-in-the-field-of-open-source-hardware-open-tools-used-for-a-security-chip/

**Comment by Syndace** [comment]
Yes, exactly, thanks for linking @mi-and-core :)

I am from [RheinMain University of Applied Sciences](https://www.hs-rm.de/en/), working at VE-HEP, where I'm responsible for the RTL-to-GDS tooling and tapeouts.

The design is a hardware security module (just a VexRiscv with a few crypto accelerators, nothing too groundbreaking but decent enough for a proof-of-concept). You can find the full design source here: https://github.com/VE-HEP/VE-HEP-HW-SW

We've taped out twice last year at IHP using OpenLane and a closed-source PDK for the sg13g2 process, and to our great pleasure, the chips are fully working! In the meantime, from within VE-HEP, we started discussion about an open PDK with IHP, which lead to IHP's open PDK we used for the tapeout this Monday :)

The ORFS design configuration, as well as all build artifacts (`flow/reports`, `flow/results` and the full GDS as taped out) are available here: https://github.com/HEP-Alliance/hsm

For any more information, feel free to contact me directly via `tim.henkes@hs-rm.de`.

Edit: repo is public now :)

**Comment by dralabeing** [comment]
Congratulations Tim @Syndace ! I will reach out to blog about this milestone.

**Comment by maliberty** [comment]
@KrzysztofHerman this is pending on your feedback.

**Comment by maliberty** [comment]
@KrzysztofHerman will you be able to review this soon?

**Comment by KrzysztofHerman** [comment]
Sorry for such a long period being silent. The parasitic extraction rules were derived using the calibration methodology provided in OpenRCX documentation. I have used Cadence QRC tool in order to derive SPEF netlist. There were some differences I have reported on Slack channel. I attach link to the [discussion](https://open-source-silicon.slack.com/archives/C0161A4A59V/p1694179971808809).

**Comment by maliberty** [comment]
I'm having difficulty following that link - could you summarize here?

**Comment by KrzysztofHerman** [comment]
@maliberty I have forwarded you the whole thread on Slack channel. Anyway, the summary is that I have made some comparison between the values of capacitances extracted using OpenRCX and QRC. The main outcome was that the vast majority of the relative difference was concentrated around 2% what, according to Tim Edwards, was quit good. Anyway there were some 10% differences as well and maybe they are the root of the calculated delays.  @Syndace, what methodology have you used to derive the rules you used for HEP project development ?  Maybe we could try some simple circuits using two rule sets and contrast the results using commercial tools ?

**Comment by Syndace** [comment]
> The parasitic extraction rules were derived using the calibration methodology provided in OpenRCX documentation.

Great. Some of the values looked like placeholders to me (a lot of "smooth" values and `0`s). Nice that I was wrong :)

> what methodology have you used to derive the rules you used for HEP project development ?

I also just followed the OpenRCX documentation, using Cadence for extraction. I still have the input patterns, the golden SPEF(s) and the scripts I used for extraction.

> Maybe we could try some simple circuits using two rule sets and contrast the results using commercial tools ?

That sounds like a great idea. You already did some sort of benchmark/comparison that gave you 2% and 10% in some cases, can you run that comparison again with my rules and maybe a few other test designs?

Here's a raw link to my rules for convenience: [link](https://raw.githubusercontent.com/The-OpenROAD-Project/OpenROAD-flow-scripts/7747f88f70daaeb63f43ce36e71829707b7e3fa7/flow/platforms/ihp-sg13g2/IHP_rcx_patterns.rules)

**Comment by maliberty** [comment]
I'm unclear on whether to proceed with this PR or close it.  What is the conclusion?  @KrzysztofHerman @Syndace

**Comment by KrzysztofHerman** [comment]
@maliberty in my opinion you can proceed with PR if we get some new data we will be back with a new PR. What is your opinion @Syndace ?

**Comment by Syndace** [comment]
Sounds sensible, let's proceed with the pr.

**Review by maliberty — APPROVED** [review]
